version: "3.9"

services:
  pihole:
    image: pihole/pihole:latest
    container_name: pihole
    hostname: pihole
    restart: unless-stopped
    networks:
      - management
    ports:
      - "53:53/tcp" 
      - "53:53/udp"
    environment:
      TZ={{TZ}}
      WEBPASSWORD={{PIHOLEWEBPASSWORD}} 
      FTLCONF_dns_upstreams=cloudflaredns#5053
    volumes:
      - pihole_etc:/etc/pihole
      - pihole_dnsmasq:/etc/dnsmasq.d
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.pihole.rule=Host(`pihole.{{HOST}}`)"
      - "traefik.http.routers.pihole.entrypoints={{TRAEFIK_ENTRYPOINT | default ('websecure')}}"
      - "traefik.http.services.pihole.loadbalancer.server.port=80"
      - "traefik.http.routers.pihole.tls={{TRAEFIK_TLS | default ('true')}}"
      - "traefik.docker.network=management"
    cap_add:
      - NET_ADMIN 
      - CHOWN
      - NET_BIND_SERVICE
      - NET_RAW
      - CAP_NET_ADMIN

networks:
  management:
    external: true

volumes:
  pihole_etc:
  pihole_dnsmasq: